lib/collection/src/shards/local_shard/snapshot.rs

```diff
@@ -108,9 +108,15 @@ impl LocalShard {
 
             if save_wal {
                 // snapshot all shard's WAL
-                Self::snapshot_wal(wal, &tar_c)
-            } else {
+                if let Some(wal) = wal {
+                    Self::snapshot_wal(wal, &tar_c)
+                } else {
+                    Ok(())
```

Comment:
I don't think just skipping WAL during the snapshot, even under read-only mode is a correct approach. The obtained snapshot will not be readable with regular deployment anymore. I think we should always copy existing WAL files, even though the WAL itself is not loaded. 

Also need to make sure `temp_path` is writable - if it is part of `./storage` - it is not